### ISSUE TITLE
org-roam-capture no longer works

### ISSUE THREAD

**Issue by jessy836** [doomemacs/doomemacs#8009]
labels: ["is:bug", "needs-triage"]

### I confirm that...
- [X] I have searched the [issue
          tracker](https://github.com/doomemacs/doomemacs/issues),
          [documentation](https://docs.doomemacs.org),
          [FAQ](https://docs.doomemacs.org/-/faq),
          [Discourse](https://discourse.doomemacs.org), and
          [Google](https://google.com), in case this issue has already been
          reported/resolved.
- [X] I have read "[How to Debug
          Issues](https://doomemacs.org/d/how2debug)", and will use it to
          provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of
          Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27,
          28, or 29. *(Unstable versions end in .50, .60, or .9x)*
### Expected behavior

When i press spc n r d t should open the file associated with today and similarly for yesterday with spc n r d y and tomorrow with spc n r d m

### Current behavior

When i press spc n r d t I get the following message:

``` emacs-lisp
apply: Wrong number of arguments: ((t) (buf) nil (save-current-buffer (set-buffer buf) (+vc-gutter--kill-thread t))), 0
```

### Steps to reproduce

      1. in init.el enable roam like this (org +roam2)
      2. doom sync
      3. doom upgrade
      4. open emacs
      5. press spc n r d t

### System Information

https://pastebin.com/7cbFRju4

**Comment by jduhamel** [comment]
I receieved the same message when trying to toggle off git-timemachine mode.

**Comment by jessy836** [comment]
> I receieved the same message when trying to toggle off git-timemachine mode.
> 
I recive also this message when spc g t to toggle  git time machine:

``` emacs-lisp
https://lists.gnu.org/archive/html/emacs-devel/2024-08/index.html
```

**Comment by EsAu79p** [comment]
This seems to be caused by ded3f5ec834942044b5caae6ca220270660b466b.
As a temporary solution this can be mitigated by **either** disabling `:ui vc-gutter` or by removing the advice defined in that commit. This should be a temporary mitigation and should be disabled when this is fixed. 

To disable `:ui vc-gutter` go to `doom-user-dir` and in `init.el` comment out the line with `vc-gutter`. `doom-user-dir` is typically `~/.config/doom` or `~/.doom.d`. 

If `vc-gutter` is needed the advice defined in the commit can be removed by adding the following to `config.el`. 

```emacs-lisp
;; FIX: `global-diff-hl-mode' enables `diff-hl-mode' *everywhere*, which calls
;;   `diff-hl-update'. If `diff-hl-update-async' is non-nil, this means a new
;;   thread is spawned for *every* buffer, whether they're visible or not. Not
;;   only can this slow a lot down, but `kill-buffer' will silently refuse to
;;   kill buffers with a thread associated with it. Chaos ensues (see #7991
;;   and #7954).
;; REVIEW: Report this upstream.
(after! diff-hl
  (defun +vc-gutter--kill-thread (&optional block?)
    (when-let ((th +vc-gutter--diff-hl-thread))
      (when (thread-live-p th)
        (thread-signal th 'quit nil)
        (when block?
          (condition-case _
              (thread-join th)
            ((quit error) nil))))))

  (defvar-local +vc-gutter--diff-hl-thread nil)
  (undefadvice! +vc-gutter--debounce-threads-a (&rest _)
    :override #'diff-hl-update
    (unless (or inhibit-redisplay
                non-essential
                delay-mode-hooks
                (null (buffer-file-name (buffer-base-buffer)))
                (null (get-buffer-window (current-buffer))))
      (if (and diff-hl-update-async
               (not
                (run-hook-with-args-until-success 'diff-hl-async-inhibit-functions
                                                  default-directory)))
          (progn
            (+vc-gutter--kill-thread)
            (setq +vc-gutter--diff-hl-thread
                  (make-thread (lambda ()
                                 (unwind-protect
                                     (diff-hl--update-safe)
                                   (setq +vc-gutter--diff-hl-thread nil)))
                               "diff-hl--update-safe")))
        (diff-hl--update))
      t))

  (undefadvice! +vc-gutter--only-tick-on-success-a (&rest _)
    :override #'diff-hl-update-once
    (unless (equal diff-hl--modified-tick (buffer-chars-modified-tick))
      (when (diff-hl-update)
        (setq diff-hl--modified-tick (buffer-chars-modified-tick)))))

  ;; HACK: This advice won't work in *all* cases (it's a C function, and any
  ;;   calls to it from C won't trigger advice), but the thread issues above are
  ;;   triggered from Elisp's buffer API (from what I can tell).
  (undefadvice! +vc-gutter--kill-diff-hl-thread-a (buf)
    :before #'kill-buffer
    (with-current-buffer buf (+vc-gutter--kill-thread t))))
```

**Comment by jessy836** [comment]
> This seems to be caused by ded3f5ec834942044b5caae6ca220270660b466b. As a
> temporary solution this can be mitigated by **either** disabling `:ui vc-gutter`
> or by removing the advice defined in that commit. This should be a temporary
> mitigation and should be disabled when this is fixed.

Thanks, for the moment I disabled vc-gutter and it work

**Comment by hlissner** [comment]
My bad. This should be fixed as of aa8c31c. Thanks for letting me know!
